assets/js/components/surveys/CurrentSurvey.js

```diff
@@ -16,32 +16,173 @@
  * limitations under the License.
  */
 
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useEffect, useState } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
 import Data from 'googlesitekit-data';
-import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import { CORE_FORMS } from '../../googlesitekit/datastore/forms/constants';
 import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
-import Survey from './Survey';
-import Portal from '../Portal';
-const { useSelect } = Data;
+import SurveyCompletion from './SurveyCompletion';
+import SurveyQuestionRating from './SurveyQuestionRating';
+const { useDispatch, useSelect } = Data;
+
+const ComponentMap = {
+	rating: SurveyQuestionRating,
+};
 
 const CurrentSurvey = () => {
-	const usingProxy = useSelect( ( select ) => select( CORE_SITE ).isUsingProxy() );
-	const currentSurvey = useSelect( ( select ) => select( CORE_USER ).getCurrentSurvey() );
+	const [ hasSentSurveyShownEvent, setHasSentSurveyShownEvent ] = useState( false );
+	const [ hasSentCompletionEvent, setHasSentCompletionEvent ] = useState( false );
+
+	useEffect( () => {
+		if ( questions?.length && ! hasSentSurveyShownEvent ) {
+			setHasSentSurveyShownEvent( true );
+			sendSurveyEvent( 'survey_shown' );
+		}
+	}, [ questions, hasSentSurveyShownEvent, sendSurveyEvent ] );
+
+	const completions = useSelect( ( select ) => select( CORE_USER ).getCurrentSurveyCompletions() );
+	const questions = useSelect( ( select ) => select( CORE_USER ).getCurrentSurveyQuestions() );
+	const surveySession = useSelect( ( select ) => select( CORE_USER ).getCurrentSurveySession() );
+
+	const formName = surveySession ? `survey-${ surveySession.session_id }` : null;
+
+	const answers = useSelect( ( select ) => select( CORE_FORMS ).getValue( formName, 'answers' ) );
+
+	const currentQuestionOrdinal = Math.max( 0, ...( answers || [] ).map( ( a ) => a.question_ordinal ) ) + 1;
+
+	const shouldHide = useSelect( ( select ) => select( CORE_FORMS ).getValue( formName, 'hideSurvey' ) );
+
+	const { setValues } = useDispatch( CORE_FORMS );
+	const { sendSurveyEvent } = useDispatch( CORE_USER );
+
+	const currentQuestion = questions?.find( ( question ) => {
+		return question.question_ordinal === currentQuestionOrdinal;
+	} );
+
+	const answerQuestion = useCallback( ( answer ) => {
+		sendSurveyEvent( 'question_answered', {
+			// eslint-disable-next-line camelcase
+			question_ordinal: currentQuestion?.question_ordinal,
+			answer,
+		} );
+
+		setValues( formName, {
+			answers: [
+				...answers || [],
+				{
+					// eslint-disable-next-line camelcase
+					question_ordinal: currentQuestion?.question_ordinal,
+					answer: { answer },
+				},
+			],
+		} );
+	}, [ answers, currentQuestion, formName, sendSurveyEvent, setValues ] );
+
+	const ctaOnClick = useCallback( () => {
+		sendSurveyEvent( 'follow_up_link_clicked' );
```

Comment:
Here the respective `completion_ordinal` should also be passed as event data, see https://docs.google.com/document/d/1jqUg4blDUFu6BCyHQVgzlz9smxK0UGrzsOpPojo2EJM/edit?pli=1#heading=h.k0jpie3yvlqo